.github/workflows/build-icon.yml

```
@@ -0,0 +1,48 @@
+name: build icon via webhook
+
+on:
+  workflow_dispatch:
+    inputs:
+      modified_components:
+        description: 'JSON array for modified components'
+        required: true
+        default: '["ALL"]'
+
+jobs:
+  build_icons:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2-beta
+        with:
+          node-version: 14
+      - name: restore lerna
+        uses: actions/cache@v2
+        with:
+          path: |
+            node_modules
+            */*/node_modules
+          key: ${{ runner.os }}-${{ hashFiles('**/yarn.lock') }}
+      - run: yarn install --frozen-lockfile
+      - run: yarn bootstrap
+      - name: Update UI catalog
+        env:
+          FILE_KEY: ${{ secrets.FIGMA_ICON_FILE_KEY }}
+          NODE_ID: ${{ secrets.FIGMA_ICON_NODE_ID }}
+          ICON_NAMES: ${{ github.event.inputs.modified_components }}
+        run: |
+          echo '{ "fileKey": "$FILE_KEY", "nodeId": "$NODE_ID", "dest": "dist", "iconNames": "$ICON_NAMES" }' > packages/spindle-icons/figma.json
```

シングルクォートで囲っているので環境変数が展開されないような気がするのですが、reg-suitみたく`$VARNAME`みたいなものでも展開されるようなコマンドでしょうか？

`get-icons.ts`を読んだ感じそうでない気がするので、nodeでJSON.stringifyするかheredocを使うのが良さそうかな〜と思いました。

```suggestion
          node -p "JSON.stringify({ fileKey: '$FILE_KEY', nodeId: '$NODE_ID', dest: 'dist', iconNames: '$ICON_NAMES' })" > packages/spindle-icons/figma.json
```

heredocの方はYAMLの文法とかの関係であまり自信ないです😂

```suggestion
          cat <<-JSON > packages/spindle-icons/figma.json
          {
             "fileKey": "$FILE_KEY",
             "nodeId": "$NODE_ID",
             "dest": "dist",
             "iconNames": "$ICON_NAMES"
          }
          JSON
```

---

完全に雑スクリプトでしたｗ nodeの方取り込みました！